src/xs2arr/model.py

```diff
@@ -115,34 +98,174 @@ def fit(
+        raise ValueError("All values in eedf_grid must be >= 0.0")
+
+    return eedf_grid
+
+
+def _validate_and_prepare_integrator(
+    integrator: str,
+) -> Callable[[np.array, np.array, np.array], float]:
+    """
+    Validates and prepares the integration method for rate computation. Returns the rate computation function.
+
+    Parameters
+    ----------
+    integrator
+        The integration method to use. Must be either 'simpson' or 'trapezoid'.
+
+    Raises
+    ------
+    TypeError
+        If integrator is not a string.
+    ValueError
+        If integrator is not 'simpson' or 'trapezoid'.
+    """
+
+    if not isinstance(integrator, str):
+        raise TypeError("integrator must be of type str")
+    if integrator not in ("trapezoid", "simpson"):
+        raise ValueError("integrator must be 'simpson' or 'trapezoid'")
+
+    return compute_rate_simpson if integrator == "simpson" else compute_rate_trapezoid
+
+
+def _validate_and_prepare_T_grid(
+    T_grid: ArrayLike | None, mean_E_grid: ArrayLike | None
+) -> np.ndarray:
+    """
+    Validates and prepares the temperature grid for rate calculations. Returns the validated temperature grid.
+
+    Parameters
+    ----------
+    T_grid
+        Temperature grid for rate calculations. If None and mean_E_grid is None,
+        a default grid of 1000 points between 0.001 and 6.0 K will be used.
+    mean_E_grid
+        Mean energy grid to define the temperature grid. If provided, T_grid will be
+        calculated as 2/3 of mean_E_grid.
+
+    Raises
+    ------
+    ValueError
+        If both T_grid and mean_E_grid are provided,
+        If provided grid is not 1-dimensional,
+        If provided grid has less than 2 points,
+        If provided grid contains negative values.
+    """
+    if T_grid is not None and mean_E_grid is not None:
+        raise ValueError("Only one of T_grid or mean_E_grid must be provided")
+    if T_grid is None and mean_E_grid is None:
+        T_grid = np.linspace(start=0.001, stop=6.0, num=1000, dtype=float)
+    if mean_E_grid is not None:
+        T_grid = 2.0 * np.asarray(mean_E_grid, dtype=float) / 3.0
+    T_grid = np.asarray(T_grid, dtype=float)
+    if T_grid.ndim != 1:
+        raise ValueError("T_grid must be 1-dimensional")
+    if len(T_grid) < 2:
+        raise ValueError("len(T_grid) must be >= 2")
+    if np.any(T_grid < 0.0):
+        raise ValueError("All values in T_grid must be >= 0.0")
+
+    return T_grid
 
 
 def _create_fitting_model(logarithmic: bool) -> tuple[FittingModel, Parameters]:
+    """
+    Creates a fitting model and its parameters for Arrhenius equation fitting. Returns the model and parameters.
+
+    Parameters
+    ----------
+    logarithmic
+        Whether to use logarithmic form of the Arrhenius equation for fitting.
+
+    Notes
+    -----
+    Default parameter values are:
+    - log10_a = -14.0 (or a = 1e-14 for non-logarithmic)
+    - b = 0.1
+    - c = -10.0 (with maximum value of 0.0)
+    """
```

Comment:
To ensure Sphinx renders this as bullet points, and to have text show up as code-blocks:

```suggestion
    Default parameter values are:

    - ``log10_a = -14.0`` (or ``a = 1e-14`` for non-logarithmic)
    - ``b = 0.1``
    - ``c = -10.0`` (with maximum value of ``0.0``)
    """
```